.github/workflows/go.yml

```diff
@@ -0,0 +1,89 @@
+name: Go
+
+on:
+  push:
+    branches: ["v2"]
+    tags:
+      - "*"
+  pull_request:
+    branches: ["v2"]
+
+jobs:
+  configure:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.18
+
+  upload:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - name: linux-x86_x64
+            os: linux
+            arch: amd64
+          - name: linux-arm64
+            os: linux
+            arch: arm64
+          - name: linux-arm32
+            os: linux
+            arch: arm
+          - name: windows-amd64
+            os: windows
+            arch: amd64
+          - name: windows-amd32
+            os: windows
+            arch: 386
+          - name: macos-intel64
+            os: darwin
+            arch: amd64
+          - name: macos-AppleSilicon
+            os: darwin
+            arch: arm64
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: create build dirs
+        run: mkdir ${{ matrix.name }}
+
+      - name: Build
+        run: GOOS=${{ matrix.os }} GOARCH=${{ matrix.arch }} go build -o ${{ matrix.name }} ./...
+
+      - name: Get Git Version String
+        working-directory: ${{github.workspace}}
+        run: |
+          if ${{ startsWith(github.ref, 'refs/tags/') }}; then
+            echo "app_version=${{ github.ref_name }}" >> $GITHUB_ENV
+          elif [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            echo "app_version=$(echo "${{ github.head_ref }}" | sed -r 's/\//_/g')-$(git rev-parse --short "${{ github.event.pull_request.head.sha }}")" >> $GITHUB_ENV
+          else
+            echo "app_version=$(git rev-parse --abbrev-ref HEAD | sed -r 's/\//_/g')-$(git describe --always)" >> $GITHUB_ENV
+          fi
+      - name: Print Git Version String
+        run: |
+          echo ${{ env.app_version }}
+      - name: Make build results executable
+        run: chmod +x mqtt-re*
+
+      - name: Zipping
+        run: zip -r ${{ matrix.name }}-${{ env.app_version }}.zip ${{ matrix.name }}
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: mqtt-record-replay-${{ matrix.name }}-${{ env.app_version }}
+          path: ${{ matrix.name }}-${{ env.app_version }}.zip
+
+      - uses: svenstaro/upload-release-action@v2
+        name: Upload binaries to release
+        if: ${{ startsWith(github.ref, 'refs/tags/') }}
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          file: ${{ matrix.name }}-${{ env.app_version }}.zip
```

Comment:
I would like to have the application name in the release zip. Either use `asset_name` or already have the right name in the zipping step.

Reply:
@jbeyerstedt What do you mean with "application name" ? can you name an example ?

Reply:
mqtt-record-replay-linux-arm64-$version.zip instead of just linux-arm64-$version.zip